### DISCUSSION TITLE
404 page with tailwind css

### DISCUSSION THREAD

**Discussion by ChrisCrossCrash** [gatsbyjs/gatsby#28883] [Help] [answered]
I'm using Gatsby with Tailwind CSS. I'm having a problem where the Tailwind styles are being applied to the development 404 page:

![image](https://user-images.githubusercontent.com/40023417/103742710-54f36900-500c-11eb-93cd-edbf74066b5b.png)

As you can see, the problem is that Tailwind removes all of the default styles. Headings, buttons, and paragraphs all look the same.

The tailwind styles are applied in `global.css`:

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

It doesn't matter if I rename the css file. The development 404 page applies whatever styles I have imported in `index.js`.

Is there any to prevent these Tailwind styles from being applied to the development 404 page? It's not such a big deal, but it would be nice to make it a little prettier.

**Comment by ramojej** [comment]
That page is not the default 404 page.. Click on the "Preview custom 404 page".

**Reply by ChrisCrossCrash** [reply]
Yes, I know that. That's why if you read my question it says "Is there any to prevent these Tailwind styles from being applied to the development 404 page?"

**Comment by robclancy** [comment]
FYI to disable the dev 404 completely you can do this in `gastby-node.js`

```js
exports.onCreatePage = ({ page, actions: { deletePage } }) => {
  if (page.path === '/dev-404-page/') {
    deletePage(page)
  }
}
```

**Comment by robclancy** [comment, marked as answer]
And just found you can make your own page to override it.

`pages/dev-404-page.tsx`. 

I copy pasted the default and made it work with tailwind. https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal-plugins/dev-404-page/raw_dev-404-page.js

**Reply by ChrisCrossCrash** [reply]
I'm not using Gatsby anymore but if it works as you say it does then that's the answer I was looking for. Thanks!

**Comment by pmagictech** [comment]
For me, I just tried to style back the elements used on the page. I choose tailwind classes that will style them close to the default style. Here is my `global.css`.
```
@tailwind base;
@tailwind components;
@tailwind utilities;

h1 {
  @apply font-bold text-3xl my-6;
}

h2 {
  @apply font-bold text-2xl my-5;
}

button, [type='submit'] {
  @apply border border-neutral-500 px-2 py-1 bg-neutral-200;
}

input {
  @apply border border-neutral-500 px-2 py-1;
}

p, pre {
  @apply my-4;
}
```

Here is what it looks like.
![image](https://github.com/gatsbyjs/gatsby/assets/31432997/03d831f3-f34b-4e5a-bc35-923e9818c6f2)
